nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp

```diff
@@ -312,6 +312,21 @@ class BtActionNode : public BT::ActionNodeBase
     auto send_goal_options = typename rclcpp_action::Client<ActionT>::SendGoalOptions();
     send_goal_options.result_callback =
       [this](const typename rclcpp_action::ClientGoalHandle<ActionT>::WrappedResult & result) {
+        if (!this->goal_handle_) {
+          if (!future_goal_handle_) {
+            // Shouldn't run into this block scope
+            RCLCPP_WARN(
+              node_->get_logger(),
+              "Goal result for %s available, but it can not get a valid goal_handle_. "
+              "Normally, this message shouldn't be printed.",
+              action_name_.c_str());
+            return;
+          }
+
+          // If the future_goal_handle_ is not nullptr, try to get goal_handle_
+          this->goal_handle_ = future_goal_handle_->get();
```

Comment:
@tzskp1
Perhaps this patch dosen't work, I got a blocking issue on this line.

Reply:
I met the blocking problem too.